geode-lucene/src/main/java/org/apache/geode/cache/lucene/internal/LuceneIndexImpl.java

```diff
@@ -78,10 +74,14 @@ public String getRegionPath() {
     return this.regionPath;
   }
 
-  protected LocalRegion getDataRegion() {
+  protected LocalRegion assignDataRegion() {
     return (LocalRegion) cache.getRegion(regionPath);
   }
 
+  protected LocalRegion getDataRegion() {
```

Comment:
could there be a race condition between the getDataRegion and the assignDataRegion... just thinking aloud. The assignDataRegion is called from afterDataRegionCreate's initialize.... but my AEQ are created in beforeDataRegionCreated and the first call in destroy commands  (like destroyAsyncEventQueue )is getDataRegion. I mean how do we handle a scenario in which a destroy index comes in between beforeDataRegionCreated and afterDataRegionCreated

Reply:
Right now destroy won't be able to get a handle to a LuceneIndexImpl until after LuceneServiceImpl.registerIndex is called.  registerIndex is called after index.initialize().

If destroy is able to get a handle to the lucene index instance before it's fully created, then we will also need to fix the destroy method, a cache lookup for a region can currently return null too.  The destroy method does not currently check for null and it doesn't need to because again, the region needs to be there before the index is fully formed.